### DISCUSSION TITLE
quick to-do list - new project

### DISCUSSION THREAD

**Discussion by luisracosta** [primefaces/community#2179] [PrimeReact Blocks]
Can you help me start a project using PrimeReact Blocks?

I followed these steps:

npx create-next-app@latest test
cd test
npm --save install primereact
npm --save install primeicons
npm --save install primeflex

and now, I have an initial project (NEXTJS react).

I have some questions:
Can I choose TS or JS? Does the library work with both?
The "copy" and "paste" blocks work with both?

I chose JS because the examples extensions are JS.

The page:
`src/app/page.js`
is the start of my test project in my project test.

What are the imports and variables to import?
I want to test this: (not all dashboards only the Sidebar dashboard)
`pages/application/dashboard/index.js`

Can you help me create a quick to-do list on how to generate a website using the dashboard and add another resource from the library?
for example an button click opens any other resource
I am having difficulty putting everything together in a new project that doesn't expose the library's code.

**Comment by vieira-brz** [comment]
Certainly! Setting up a Next.js project with PrimeReact and creating a dashboard with additional functionalities like buttons opening resources can be exciting. Let’s break down the process step-by-step, including the use of JavaScript (JS), which you’ve chosen based on the example extensions.

### Step 1: Project Setup with PrimeReact
You've already initialized your Next.js project and installed PrimeReact, PrimeIcons, and PrimeFlex. Here’s a quick checklist to ensure everything is set up correctly:
- **Next.js app initialization**: `npx create-next-app@latest test`
- **Navigate into the project**: `cd test`
- **Install PrimeReact and related libraries**:
  ```bash
  npm install primereact primeicons primeflex
  ```

### Step 2: Choosing Between TypeScript (TS) and JavaScript (JS)
PrimeReact works seamlessly with both TypeScript and JavaScript. You can choose based on your preference or project requirements:
- **TypeScript**: Offers type safety and is beneficial for larger projects or projects where you need to manage complex data structures.
- **JavaScript**: Easier to start with and sufficient for many projects, especially if you're more familiar with it.

Since you have chosen JavaScript, we will continue with JS.

### Step 3: Basic Setup for Using PrimeReact
To use PrimeReact components, you need to import the necessary CSS files in your project. You can add these lines to your `pages/_app.js` to include global styles:

```javascript
import 'primereact/resources/themes/saga-blue/theme.css';  //theme
import 'primereact/resources/primereact.min.css';          //core css
import 'primeicons/primeicons.css';                        //icons
```

### Step 4: Creating a Dashboard with Sidebar
You want to start with a dashboard layout using a sidebar. Let’s set up a basic dashboard page:

1. **Create a new file for your dashboard** at `pages/dashboard.js`.
2. **Import PrimeReact components** you plan to use, starting with the Sidebar:
   ```javascript
   import { Sidebar } from 'primereact/sidebar';
   import { Button } from 'primereact/button';
   ```

3. **Set up the Sidebar component** within your dashboard:
   ```javascript
   import React, { useState } from 'react';
   import { Sidebar } from 'primereact/sidebar';
   import { Button } from 'primereact/button';

   const Dashboard = () => {
       const [visible, setVisible] = useState(false);

       return (
           <>
               <Button label="Toggle Sidebar" onClick={(e) => setVisible(true)} />
               <Sidebar visible={visible} onHide={() => setVisible(false)}>
                   <h3>Dashboard Content</h3>
                   {/* Add other components or links here */}
               </Sidebar>
           </>
       );
   }

   export default Dashboard;
   ```

### Step 5: Adding Another Feature (Button to Open a Resource)
Let’s say you want a button that opens a dialog or another component. You could integrate the `Dialog` component from PrimeReact:

1. **Add the Dialog import** to your `dashboard.js`:
   ```javascript
   import { Dialog } from 'primereact/dialog';
   ```

2. **Setup state and Dialog** in your component:
   ```javascript
   const [dialogVisible, setDialogVisible] = useState(false);

   // Inside your return statement, add a button and Dialog component
   <Button label="Open Dialog" onClick={() => setDialogVisible(true)} />
   <Dialog header="Dialog" visible={dialogVisible} onHide={() => setDialogVisible(false)}>
       <p>This is a simple dialog component.</p>
   </Dialog>
   ```

### Step 6: Running and Testing Your Application
- Run your application with `npm run dev` and navigate to `http://localhost:3000/dashboard` to see your new dashboard with the Sidebar and additional functionalities.

### Final Notes
- **Experiment and expand**: As you grow comfortable, you can add more components and utilities from PrimeReact.
- **Consider state management**: For a more complex application, you might want to look into state management solutions like Redux or Context API.

This setup should give you a good starting point for creating a dashboard with a sidebar and integrating additional PrimeReact components into a Next.js project using JavaScript.

**Comment by luisracosta** [comment]
Thank you very much for the support.
I purchased primeUI blocks react.
Is the procedure the same?
